Dockerfile

```diff
@@ -22,29 +22,47 @@ ARG SKIP_MINIFY
 
 EXPOSE 8888
 
+# Set the default environment to be CI, can be overridden by passing new CONFIG_ENV setting at container start
+ENV CONFIG_ENV ci
+ENV DOCKER_CONTAINER true
+
 # Copy in the narrative repo
-ADD ./ /kb/dev_container/narrative
 ADD ./kbase-logdb.conf /tmp/kbase-logdb.conf
+ADD ./narrative-log-proxy.conf /kb/dev_container/narrative/
 ADD ./deployment/ /kb/deployment/
-WORKDIR /kb/dev_container/narrative
+ADD ./build.js /kb/dev_container/narrative/
```

Comment:
Copying over the directories and files that are actually used, instead of everything under the sun. Attempted to order them by approximate likelihood of being changed, although everything after nbextensions is a lot more likely to be edited than the stuff before it.

Reply:
Good idea. Eventually (not this PR, and maybe not right away) it might be nice to make a production-only image and leave out tests, test data, test-only dependencies, and so on.

But this is a good step on that path!